### PR TITLE
Add permissions-policy check for publicKey credentials

### LINKED ISSUE #136: add permissions policy support for webauthn
webauthn has needs to integrate with [Feature Policy](https://www.w3.org/TR/feature-policy/): https://github.com/w3c/webauthn/issues/911 "integrate with feature policy..."

This issue is only to note the above. There are overall questions wrt whether we do credential feature polic{y,ies} as per-credential or all-creds-included: see issue #135.

### PR DESCRIPTION
**Pull request by equalsJeffH**
If the `CredentialRequestOptions` supplied to the Request a Credential algorithm contains a object named `publicKey` then check that the responsible document is allowed to use the publickey-credentials-get policy-controlled feature.

This PR supersedes PR #138.

Fixes #136 
Improves #135

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-credential-management/pull/182.html" title="Last updated on Jan 6, 2022, 9:42 PM UTC (e915c78)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-credential-management/182/b273b4a...e915c78.html" title="Last updated on Jan 6, 2022, 9:42 PM UTC (e915c78)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (nsatragno) [inline index.bs:788]: We should add web OTP

https://wicg.github.io/web-otp/#sctn-permissions-policy

This mechanism is a bit of a pain to extend to other credential types and somewhat defeats the purpose of the credential types registry. Can we add the permission policy names to the registry, and then iterate those?

File: index.bs
Comment (equalsJeffH) [reply index.bs:11]: yep, thx, good catch!

File: index.bs
Comment (jyasskin) [inline index.bs:791]: You don't really need the `[[PERMISSIONS-POLICY]]` citation since `[=policy-controlled feature=]` links directly there and makes sure it's included in the references.
```suggestion
        [=policy-controlled feature=] return [=a promise rejected with=] `NotSupportedError`.
```

Comment (jyasskin) [reply index.bs:791]: https://github.com/w3c/webappsec-permissions-policy/issues/396 suggests `NotAllowedError` or consistency with other exceptions thrown by the interface. The PasswordCredential already throws a `NotAllowedError`, while PublicKeyCredentials throw both that and `NotSupportedError`. Altogether, I think that implies

```suggestion
        [=policy-controlled feature=] [[PERMISSIONS-POLICY]] return [=a promise rejected with=] a "{{NotAllowedError}}" {{DOMException}}.
```

File: index.bs
Comment (equalsJeffH) [reply index.bs:786]: doh! But of course, we just need to use:
> If options[publicKey] [exists](https://infra.spec.whatwg.org/#map-exists)...

File: index.bs
Comment (equalsJeffH) [reply index.bs:788]: Good idea, agreed. Although I suggest we:
1. submit an issue about this.   ( done: see issue #184 )
2. address it in a separate PR **_after both_** this PR and [PR 181](https://github.com/w3c/webappsec-credential-management/pull/181) (which creates the registry) land.

File: index.bs
Comment (equalsJeffH) [reply index.bs:791]: Ok, thx, good suggestions, I've incorporated them by hand since the github UI was acting weird (I think because both suggestions are on the same line) and I was not sure what it was going to do.

**Comment by equalsJeffH** [comment]
I am thinking that we ought to land this PR before we land PR #181 so the latter PR can incorporate these changes (by rebasing onto main branch) because the changes in this PR are less invasive that those in PR 181.

**Comment by equalsJeffH** [comment]
I resolved the conflicts with main branch using the github UI (which did a merge-from-main it seems (sorry Nina if that was the wrong thing to do).

I think this PR is ready to merge modulo the three outstanding questions (above) I have for @nina @jyasskin:
https://github.com/w3c/webappsec-credential-management/pull/182/#discussion_r773393848 fix issue #184 in separate PR ?
https://github.com/w3c/webappsec-credential-management/pull/182/#discussion_r773415912 which error to return is not allowed by policy?
https://github.com/w3c/webappsec-credential-management/pull/182/#issuecomment-999091144  Merge this PR before PR #181 ?

File: index.bs
Comment (jyasskin) [reply index.bs]: I'm not familiar enough with the implementation to confidently confirm that, but I do read the code the same way you did.

File: index.bs
Comment (jyasskin) [inline index.bs:837]:
```suggestion
    1.  If |options|[{{CredentialRequestOptions/publicKey}}] [=map/exists=] then
```

File: index.bs
Comment (jyasskin) [inline index.bs:73]: https://infra.spec.whatwg.org/#map-exists is exported, so you don't need to define it locally if you use the right `[=for/=]` syntax below.

```suggestion
```

File: index.bs
Comment (jyasskin) [inline index.bs:839]: There's a [shorthand](https://tabatkins.github.io/bikeshed/#autolink-inside) for this, but you don't have to use it:

```suggestion
        [=publickey-credentials-get-feature|publickey-credentials-get=]
```

**Review by jyasskin — APPROVED** [review]
Merging into your feature branch won't cause any problems since the repo is configured to only allow squash-merges back to `main`.

File: index.bs
Comment (kenrb) [reply index.bs]: Yes that is correct.

Comment (equalsJeffH) [reply index.bs]: Ok, thanks folks, this question is now confirmed, am resolving this conversation.

**Comment by equalsJeffH** [comment]
Thanks @jyasskin for the corrections.  I think this PR is now ready to land.

**Pull request merged**
